src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -452,22 +445,22 @@ public PrimaryKeys apply(PrimaryKeys existing, PrimaryKey neww)
     static class PrimaryKeysRemover implements MemtableTrie.UpsertTransformer<PrimaryKeys, PrimaryKey>
     {
         private final LongAdder heapAllocations;
-        private Object ref;
+        private Set<PrimaryKeys> modifiedPrimaryKeys;
 
         PrimaryKeysRemover(LongAdder heapAllocations)
         {
             this.heapAllocations = heapAllocations;
         }
 
         /**
-         * Set the reference object to be used in the apply method.
+         * Set the set of modifiedPrimaryKeys.
          * Warning: This method is not thread-safe and should only be called from within the synchronized block
          * of the TrieMemoryIndex class.
-         * @param ref
+         * @param modifiedPrimaryKeys
          */
-        private void setReference(Object ref)
+        private void setModifiedPrimaryKeys(Set<PrimaryKeys> modifiedPrimaryKeys)
```

Comment:
Nit: This name seems to make sense only when looking at this class from the outside perspective. The word "modified" applies to the external context, the context of the caller. However, looking at the stuff this class does, it doesn't convey any meaning - it's even a bit confusing. So if I e.g. had this class in a third-party library (let's assume for a moment the external stuff doesn't exist), I'd have no idea what this set really means and how it affects the behavior of this remover. Can we just rename it to something like `ignoredPrimaryKeys`, or `primaryKeysToIgnore` and update the description that this remover will *not* remove the keys from this set when asked to?

Reply:
I agree the name isn't great as is.

Reply:
I went with "seen" because we use it in two places, one to know if we've seen it before and another to ignore (because we saw it).